### ISSUE TITLE
Noble required services at boot (RPi 3)

### ISSUE THREAD

**Issue by freebeans** [sandeepmistry/noble#523]
I'm using noble on a RPi 3 and my script is executed during the boot process.
AFAIK, Raspbian uses systemd with legacy init scripts support, so I configured my script to boot after the bluetooth service.

I'm using part of your example modified as follows:
```
noble.on('stateChange', function(state) {
	if (state === 'poweredOn') {
		noble.startScanning([], true);
		console.log("[BLE ] NOBLE started scanning");
	} else {
		noble.stopScanning();
		console.log("[BLE ] NOBLE stopped scanning");
	}
});
```
I'm piping the output to a log file. My script runs at boot but the "NOBLE started scanning" message never appears and the script doesn't scan beacons. Running the script on a shell works just fine. Adding a delay (`sleep 5`) to the init script makes everything work, so I guess my script should wait for some more services before it is executed.

Which services does noble require to be running other than bluetooth? I see there's one called hciuart.

**Comment by sandeepmistry** [comment]
@freebeans 

> Which services does noble require to be running other than bluetooth? I see there's one called hciuart.

Yes, after the hciuart ...

**Comment by vaquierm** [comment]
Hi I am currently encountering the same issue using a beagleBoard.
However even if I add the delay, I never see the statechange. I am printing noble.state at a small interval and what i see in my logfile is just a series of "poweredOff". however if I restart the script, everything works fine. It seep to only be a problem when the device boots up. (which is a huge issue since the whole point was for it to work right away witout any manual input). I am using <https://github.com/chovy/node-startup> for the boot up script.
Would you know why noble does not trigger this statechange?

**Comment by freebeans** [comment]
Maybe you can try to add both **hciuart** and **bluetooth** as dependencies of your init script.
You can also consider increasing the delay, but if you're using sysvinit, the script will hold the boot process and prevent other services to start running, and increasing the delay will only increase boot time.

tl;dr: you should include all dependencies on your init script.

Let me know if it helps you.

**Comment by vaquierm** [comment]
Thank you for your response, I am a beginner in all this. What do you man by adding them as dependencies of my init script, and how would I go about that?

But if I understand correctly, If I put a delay long enough for my node app to actually start running, it should work fine right? Because that doesn't seem to work either. The app kept printing "poweredOff" and the moment I use the **restart** command of the node-app script from <https://github.com/chovy/node-startup>, it doesn't even log "poweredOff" a single time, it is like noble has magically powered on just with the action of restarting the app.
